packages/store/src/keys/compareKeys.js

```diff
@@ -99,11 +101,12 @@ export const compareKeys = (left, right) => {
         }
       }
       // If copyRecord X is smaller than copyRecord Y, then they must have the
-      // same property names, and every value is X must be smaller or equal to
-      // every corresponding value in Y. The rank order of X and Y is based
-      // on lexicographic rank order of their values, as organized by the
-      // reverse order of their property names. Thus
-      // if compareKeys(X,Y) < 0 then compareRank(X,Y) <= 0.
+      // same property names and every value in X must be smaller or equal to
+      // the corresponding value in Y (with at least one value smaller).
```

Comment:
Note the caveat here. If every value is equal, then X is _not_ smaller than Y (because they are equivalent).